theopensystemslab/planx-new
# feat: send a Slack notification when flow status is updated on production

Replaces #3466 

Now a much simpler approach than trying to hook into existing `/webhooks/hasura/send-slack-notification` & submission events !

**Changes:**
- Adds a new API endpoint which can be triggered by `teamEditors` (and `platformAdmins` implicitly) which accepts a simple `message` body and posts to #planx-notifications if we're on production env
- Calls this endpoint from flow status `onSubmit` if "save" has been successful

**To test:**
- Navigate to flow settings -> flow status in the editor on this pizza & open your "Network" tab
- Toggle the flow status and confirm a request has kicked off with response `200` - message should indicate it's skipping the actual Slack post because not production, but still display the message that _would have_ posted - confirm the team, flow slug and status are correct
  - If you're in a team that we want to skip (eg "testing"), confirm that no notification request kicks off at all but you can still update your flow status as normal

**Open questions** noted in original PR still apply here, will raise to Silvia before merging as she's main audience of this! :heavy_check_mark: Answers in thread here https://opensystemslab.slack.com/archives/C5Q59R3HB/p1724140076950159
